core/js/src/main/scala/org/http4s/js/crypto/Crypto.scala

```diff
@@ -0,0 +1,937 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Is this original work?  If not, can we attribute it and update NOTICES?  

If so, good grief, I'm sorry for the yaks you must shave.  Is this something that should be a library somewhere?  Or made `private[http4s]` so it's not an API liability for something that isn't actually directly HTTP?  That's what our internal package is about.

It is stupid o'clock here, and I am not qualified to review this package even at smart o'clock.  Is there anyone we can bring in?

Reply:
This is copy-pasta from https://github.com/scala-js/scala-js-dom/ (the `for3Use2_13` from that came up before :P). I'll fix the attribution/NOTICES.

Basicaly it's a really terrible situation where this identical API is available on Browsers and Node.js (see also https://github.com/http4s/http4s/pull/4938#discussion_r683102955) but the facades are written in such a way that they explode if they are on the wrong platform. You can read my story about this one here: https://users.scala-lang.org/t/scala-js-how-to-support-both-node-js-and-browser-platforms/7623

Reply:
Also, yes we should probably make the entire `js` package internal. At the time I was wondering if downstreams from http4s would want these utilities, but better not be on the hook as you say! Fortunately facades are cheap.

Reply:
Darn, we can't make just make this internal, I was being silly. At least some of these APIs _do_ need to be public (e.g., a `SecretKey` class for `CSRF`). I think that we're going to be on the hook for these after all :(